src/config.rs

```diff
@@ -131,38 +122,145 @@ pub struct Opt {
 
     /// The RRDP notification file URI of the Krill instance to sync with
     pub notification_uri: Https,
+
+    /// An optional mapping of the notification file URI path up to the parent directory of
+    /// the notification file itself, to a different base URI / path on disk.
+    ///
+    /// Note: For Krill and RIPE NCC - snapshots and deltas can always be found in subdirectories
+    ///        **under** the base dir of the notification file. However, we may want to revisit this
+    ///        configuration in future to allow for explicit mapping of multiple URI paths.
+    #[structopt(
+        long = "source_uri_base",
+        value_name = "alternate fetch base uri or dir for notification uri"
+    )]
+    pub source_uri_base: Option<FetchSource>,
+
+    #[structopt(skip)]
+    pub fetch_map: Option<FetchMap>,
 }
 
-fn log_without_target(
-    out: fern::FormatCallback,
-    message: &std::fmt::Arguments,
-    record: &log::Record,
-) {
-    out.finish(format_args!(
-        "{} {}: {}",
-        chrono::Local::now().format("%Y/%m/%d %H:%M:%S"),
-        record.level(),
-        message,
-    ))
+impl Config {
+    pub fn rsync_enabled(&self) -> bool {
+        !self.rsync_disable
+    }
+
+    pub fn rsync_dir_use_symlinks(&self) -> bool {
+        if cfg!(unix) {
+            !self.rsync_dir_force_moves
+        } else {
+            false
+        }
+    }
+
+    pub fn rsync_dir_current(&self) -> PathBuf {
+        self.rsync_dir.join("current")
+    }
+
+    pub fn fetcher(&self) -> Fetcher {
+        Fetcher::new(self.notification_uri.clone(), self.fetch_map.clone())
+    }
+
+    pub fn state_path(&self) -> PathBuf {
```

Comment:
`s/fn state_path()/fn rrdp_state_path()/` to make it clear that it is about RRDP state only, not state in general, and to be more consistent with `fn rsync_state_path()`.

Reply:
ack